crates/ruff/src/rules/ruff/rules/implicit_optional.rs

```diff
@@ -141,6 +158,18 @@ impl<'a> Iterator for PEP604UnionIterator<'a> {
     }
 }
 
+fn is_known_type(call_path: &CallPath) -> bool {
+    match call_path.as_slice() {
+        ["typing" | "typing_extensions" | "types", _] => true,
+        _ => {
+            is_standard_library_generic(call_path)
+                || is_pep_593_generic_type(call_path)
+                || is_immutable_non_generic_type(call_path)
+                || is_immutable_generic_type(call_path)
+        }
+    }
+}
```

Comment:
\cc @charliermarsh I'm a bit torn on this. Is this enough or too much? Should we just go with either builtin or coming from known module which contains types? There are others apart from `typing` or `typing_extensions` such as `collections`, `types`, and even `re`, `io`, `contextlib`, etc. These are covered in the above functions at a granular level.

Reply:
Hmm, I feel like we _could_ accept anything that's builtin, or from `typing_extensions`, or _any_ standard-library module. Should we just check in `KNOWN_STANDARD_LIBRARY`?

Reply:
Yeah, `KNOWN_STANDARD_LIBRARY` makes more sense.